wasm/arguments.go

```diff
@@ -106,3 +106,19 @@ func NewStoreWriterOutput(name string, store store.Store, updatePolicy pbsubstre
 		ValueType:    valueType,
 	}
 }
+
+type ParamsInput struct {
+	BaseArgument
+	BaseValueArgument
+}
+
+func NewParamsInput(name string, value string) *ParamsInput {
+	return &ParamsInput{
+		BaseArgument: BaseArgument{
+			name: name,
+		},
+		BaseValueArgument: BaseValueArgument{
+			value: []byte(value),
+		},
+	}
```

Comment:
That whole `BaseArgument` I think is used only for logging purposes (and even then, if you track it down, that `from string` thing is _not_ even used in `WriteAtPtr()`). So you can make `BaseArgument{name: "params"}` and that'll be fine.
Which means you need only _one_ param here, the `value`.